app/assets/javascripts/analytics.js

```diff
@@ -1,5 +1,85 @@
 /* global $ */
 
 $(function(){
+  'use strict';
 
-});
\ No newline at end of file
+  var ifPage = function (klass, f){
+    if( $(document.body).hasClass(klass) && f instanceof Function ){
+      f();
+    }
+  }
+
+  var sendEvent = function ( selector, ev, page, eventLabel, text ){
+    $(selector)[ev](function (){
+        // ga('send', 'event', 'search-index-page', 'next-button', 'Next button clicked to start');
+        if( ga && ga instanceof Function)
+          ga('send', 'event', page, eventLabel, text);
+    });
+  }
+
+
+  ifPage('login-page', function (){
+    sendEvent('input.button', 'click', 'login-page', 'login-with-email', 'Clicked on "Log in with email" button.');
+    sendEvent('a.login-button', 'click', 'login-page', 'login-with-google', 'Clicked on "Log in with Google" button.');
+  });
+
+
+  ifPage('home-page', function (){
+    ga('send', 'event', 'home-page', 'looking-at-home-page', 'User is looking at the home page');
+
+    sendEvent('form.search-box', 'submit', 'home-page', 'home-page-search', 'User made a search from the home page');
+    sendEvent('.leaf-node a.subteam-link', 'click', 'home-page', 'home-page-used-the-org-browser', 'User went to a team page from the org browser');
+    sendEvent('.add-new-person', 'click', 'home-page', 'home-page-add-new-profile', 'User clicked on add new profile');
+
+    $('.new-org-browser').click(function (ev){
+      var $self = $(ev.target).closest('.new-org-browser');
+      if( !$self.hasClass('tracked-for-click') ){
+        ga('send', 'event', 'home-page', 'home-page-explored-the-org-browser', 'User clicked on the org browser');
+        $self.addClass('tracked-for-click');
+      }
+    });
+  });
+
+
+  ifPage('search-page', function (){
+    ga('send', 'event', 'search-page', 'looking-at-search-page', 'User is looking at the search page');
+
+    var values = [
+      { label: 'search-page-0-results', text: 'No results found for keyword' },
+      { label: 'search-page-1-result',  text: '1 result found for keyword' },
+      { label: 'search-page-2-results', text: '2 results found for keyword' }
+    ];
+
+    var multiValue = {
+      label: 'search-page-multiple-results', text: 'More than two results found for keyword'
+    };
+
+    var resultCount = $('.search-result').length;
+
+    if( resultCount < 3 ){
+      ga('send', 'event', 'search-page', values[resultCount].label, values[resultCount].text);
+    } else {
+      ga('send', 'event', 'search-page', multiValue.label, multiValue.text);
```

Comment:
We had a chat with @threedaymonk about this and reckoned this doesn't need more abstracting for now.